### DISCUSSION TITLE
Should we start adding type hinting?

### DISCUSSION THREAD

**Discussion by phil-bell** [encode/django-rest-framework#9581] [Ideas & Suggestions] [answered]
**Description:**
At the moment the isn't any [type hinting](https://docs.python.org/3/library/typing.html) in the code base. I wanted to open up the discussion on if we should start adding it?

**Problem:**
There are some situations where the lack of type checking confuses IDE linters and results in them warning users of none issues. 
For example:
Using the `get_serializer` method on `GenericAPIView` results in the linter thinking the code will not continue
![image](https://github.com/user-attachments/assets/3238f44a-cbd7-4ac5-89e1-47100d13f314)
Linter message:
![image](https://github.com/user-attachments/assets/54d3b437-78d0-4ec2-8e06-42093c706cd7)

**Comment by TGoddessana** [comment, marked as answer]
you can use [djangorestframework-stubs](https://github.com/typeddjango/djangorestframework-stubs) library, which adds type stubs for django and drf.

As a side note, the [django team has decided to wait until Python's type hinting ecosystem is more stable.](https://groups.google.com/g/django-developers/c/C_Phs05kL1Q/discussion)
